src/modules/simulation/gz_bridge/GZGimbal.cpp

```diff
@@ -151,11 +163,14 @@ bool GZGimbal::pollSetpoint()
 		gimbal_controls_s msg;
 
 		if (_gimbal_controls_sub.copy(&msg)) {
-			// map control inputs from [-1;1] to [min_angle; max_angle] using the range parameters
-			_roll_stp = math::constrain(math::radians(msg.control[msg.INDEX_ROLL] * _mnt_range_roll / 2), _roll_min, _roll_max);
-			_pitch_stp = math::constrain(math::radians(msg.control[msg.INDEX_PITCH] * _mnt_range_pitch / 2), _pitch_min,
-						     _pitch_max);
-			_yaw_stp = math::constrain(math::radians(msg.control[msg.INDEX_YAW] * _mnt_range_yaw / 2), _yaw_min, _yaw_max);
+			// map control inputs from [-1;1] to [min_angle; max_angle]
+			_roll_stp = math::constrain(math::radians(msg.control[msg.INDEX_ROLL] * ((_mnt_max_roll - _mnt_min_roll) / 2 + _mnt_off_roll)),
```

Comment:
Don't you need to distinguish positive and negative control input cases?
if(u is neagive) 
 angle=u"_mnt_min_roll + offset
if(positive
  angle=u"_mnt_max_roll + offset